models/rubicon/base/rubicon_base_trades.sql

```diff
@@ -0,0 +1,115 @@
+    schema = 'rubicon_base',
+    tags = ['dunesql'],
+    alias = alias('trades'),
+    partition_by = ['block_month'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index', 'trace_address'],
+    post_hook='{{ expose_spells(\'["base"]\',
+                                "project",
+                                "rubicon",
+                                \'["denver"]\') }}'
+    )
+}}
+
+{% set project_start_date = '2023-08-08' %}
+
+WITH dexs AS
+(
+    -- -- useful syntax when dealing with the event LogTake
+    -- -- pay_gem corresponds with take_amt - this is what the taker is taking and what the maker is selling
+    -- -- buy_gem corresponds with give_amt - this is what the taker is giving and what the maker is buying
+
+    --From the prespective of the taker
+    SELECT
+        t.evt_block_time AS block_time, 
+        t.evt_block_number,
+        t.taker AS taker,
+        t.maker AS maker,
+        t.take_amt AS token_bought_amount_raw,
+        t.give_amt AS token_sold_amount_raw,
+        cast(NULL as double) AS amount_usd,
+        t.pay_gem AS token_bought_address,
+        t.buy_gem AS token_sold_address,
+        t.contract_address as project_contract_address,
+        t.evt_tx_hash AS tx_hash,
+        '' AS trace_address,
+        t.evt_index
+    FROM
+        {{ source('rubicon_base', 'RubiconMarket_evt_emitTake') }} t
+        
+    WHERE t.evt_block_time >= cast('{{ project_start_date }}' AS timestamp)
+    {% if is_incremental() %}
+    AND t.evt_block_time >= date_trunc('day', now() - interval '7' day)
+    {% endif %}
+)
+SELECT
+    'base' AS blockchain,
+    'rubicon' AS project,
+    '1' AS version,
+    TRY_CAST(date_trunc('DAY', dexs.block_time) AS date) AS block_date,
+    CAST(date_trunc('month', dexs.block_time) AS date) AS block_month,
+    dexs.block_time,
+    erc20a.symbol AS token_bought_symbol,
+    erc20b.symbol AS token_sold_symbol,
+    case
+        when lower(erc20a.symbol) > lower(erc20b.symbol) then concat(erc20b.symbol, '-', erc20a.symbol)
+        else concat(erc20a.symbol, '-', erc20b.symbol)
+    end as token_pair,
+    dexs.token_bought_amount_raw / power(10, erc20a.decimals) AS token_bought_amount,
+    dexs.token_sold_amount_raw / power(10, erc20b.decimals) AS token_sold_amount,
+    dexs.token_bought_amount_raw AS token_bought_amount_raw,
+    dexs.token_sold_amount_raw AS token_sold_amount_raw,
+    coalesce(
+        dexs.amount_usd,
+        (dexs.token_bought_amount_raw / power(10, p_bought.decimals)) * p_bought.price,
+        (dexs.token_sold_amount_raw / power(10, p_sold.decimals)) * p_sold.price
+    ) AS amount_usd,
+    dexs.token_bought_address,
+    dexs.token_sold_address,
+    dexs.taker,
+    dexs.maker,
+    dexs.project_contract_address,
+    dexs.tx_hash,
+    tx."from" AS tx_from,
+    tx.to AS tx_to,
+    dexs.trace_address,
```

Comment:
we can remove `trace_address`

Reply:
i have gone through and removed `trace_address` from `rubicon.trades`. its worth noting, i kept `trace_address` on any legacy file, but have removed it from the schema. somewhat confused on how this doesn't cause an issue...

Reply:
leaving in legacy is fine. we're only doing it for looking forward. the schema file is simply for documentation purposes (other than the tests), so it doesn't hurt to keep it, but will write false info to the dbt docs (such as say this spell includes trace_address, when really it doesn't)